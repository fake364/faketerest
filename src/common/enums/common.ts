export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
