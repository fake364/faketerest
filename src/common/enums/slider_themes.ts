export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
