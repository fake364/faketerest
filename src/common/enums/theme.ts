export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}