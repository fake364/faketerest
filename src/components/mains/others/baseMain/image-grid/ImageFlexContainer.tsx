import React, { useRef } from 'react';
import { CommonTypes } from '../../../../../common/types/common';
import CommonUtils from '../../../../../common/utils/creation-utils/arrays/common';
import { chunkImageIndexes, getImageById } from './flex-cell/utils/utils';
import ImageCell from './flex-cell/ImageCell';
import { SLIDER_ARRAY } from '../../../../../common/classes/customClasses/SliderNode';
import clsx from 'clsx';

const margins = ['', 'mt-[5rem]', 'mt-[10rem]', 'mt-[15rem]'];

const delays = [
  'opacity_move_up-0',
  'opacity_move_up-250',
  'opacity_move_up-500',
  'opacity_move_up-725',
  'opacity_move_up-1000',
  'opacity_move_up-1250',
  'opacity_move_up-1500'
];

const ImageFlexContainer: React.FC<CommonTypes.ShownType> = ({ shownType }) => {
  const { current: indexArray } = useRef(chunkImageIndexes(34, 5));
  const { current: marginPalindrome } = useRef(
    CommonUtils.numericPalindrome(3, 0)
  );
  return (
    <div className="slider-flex-container">
      {indexArray.map((columnIndexes, columnIndex) => (
        <div
          className={clsx(
            'flex flex-col flex-1 relative',
            margins[marginPalindrome[columnIndex]],
            'opacity_move_up',
            delays[columnIndex]
          )}
        >
          {columnIndexes.map((imageIndex) => {
            const images = SLIDER_ARRAY.map((theme) =>
              getImageById(theme, imageIndex)
            );
            return <ImageCell shownType={shownType} images={images} />;
          })}
        </div>
      ))}
    </div>
  );
};

export default ImageFlexContainer;
