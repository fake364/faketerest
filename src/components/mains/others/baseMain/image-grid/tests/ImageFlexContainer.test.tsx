import React from 'react';
import { shallow } from 'enzyme';
import ImageFlexContainer from '../ImageFlexContainer';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';

describe('ImageFlexContainer', () => {
  it('should render container of images with margins and delays', () => {
    const wrapper = shallow(
      <ImageFlexContainer
        isAnimation={true}
        shownType={SLIDER_THEMES.HOME_DECOR}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
