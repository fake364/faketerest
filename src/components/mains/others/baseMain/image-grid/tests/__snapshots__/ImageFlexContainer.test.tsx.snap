// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFlexContainer should render container of images with margins and delays 1`] = `
<div
  className="slider-flex-container slider-gradient"
>
  <div
    className="flex flex-col flex-1 relative opacity_move_up opacity_move_up-0"
    key="0"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative mt-[5rem] opacity_move_up opacity_move_up-250"
    key="1"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative mt-[10rem] opacity_move_up opacity_move_up-500"
    key="2"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative mt-[15rem] opacity_move_up opacity_move_up-725"
    key="3"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative mt-[10rem] opacity_move_up opacity_move_up-1000"
    key="4"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative mt-[5rem] opacity_move_up opacity_move_up-1250"
    key="5"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="4"
      shownType="homeDecor"
    />
  </div>
  <div
    className="flex flex-col flex-1 relative opacity_move_up opacity_move_up-1500"
    key="6"
  >
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="0"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="1"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="2"
      shownType="homeDecor"
    />
    <ImageCell
      images={
        Array [
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
          Object {
            "process": [Function],
          },
        ]
      }
      key="3"
      shownType="homeDecor"
    />
  </div>
</div>
`;
